tee-worker/litentry/core/assertion-build/src/lit_staking.rs

```diff
@@ -0,0 +1,235 @@
+	params: Vec<String>,
+}
+
+impl JsonRPCRequest {
+	pub fn state_getstorage(params: &str) -> Self {
+		Self {
+			id: 1,
+			jsonrpc: "2.0".to_string(),
+			method: "state_getStorage".to_string(),
+			params: vec![params.to_string()],
+		}
+	}
+}
+
+impl RestPath<String> for JsonRPCRequest {
+	fn get_path(path: String) -> core::result::Result<String, RestClientError> {
+		Ok(path)
+	}
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct JsonRPCResponse {
+	pub id: usize,
+	pub jsonrpc: String,
+	pub result: String,
+}
+
+pub struct LitentryStakingClient {
+	client: RestClient<HttpClient<DefaultSend>>,
+}
+
+impl Default for LitentryStakingClient {
+	fn default() -> Self {
+		Self::new()
+	}
+}
+
+impl LitentryStakingClient {
+	pub fn new() -> Self {
+		let mut headers = Headers::new();
+		headers.insert(CONNECTION.as_str(), "close");
+		let client = build_client("https://litentry-rpc.dwellir.com:443", headers);
+		LitentryStakingClient { client }
+	}
+
+	pub fn send_request(&mut self, data: &JsonRPCRequest) -> Result<JsonRPCResponse> {
+		self.client
+			.post_capture::<String, JsonRPCRequest, JsonRPCResponse>(String::default(), data)
+			.map_err(|e| {
+				Error::RequestVCFailed(
+					Assertion::LITStaking,
+					ErrorDetail::DataProviderError(ErrorString::truncate_from(
+						format!("{e:?}").as_bytes().to_vec(),
+					)),
+				)
+			})
+	}
+}
+
+pub trait QueryParachainStaking {
+	fn query_delegator_state(&mut self, key: &str) -> Result<String>;
+}
+
+impl QueryParachainStaking for LitentryStakingClient {
+	fn query_delegator_state(&mut self, key: &str) -> Result<String> {
+		let data = JsonRPCRequest::state_getstorage(key);
+		let res = self.send_request(&data)?;
+		Ok(res.result)
+	}
+}
+
+pub struct DelegatorState;
+impl DelegatorState {
+	pub fn query_lit_staking(
+		&mut self,
+		client: &mut LitentryStakingClient,
+		identities: &[Identity],
+	) -> Result<u128> {
+		let mut total_staking_amount = 0_u128;
+
+		for identity in identities {
+			let storage_key = DelegatorState::delegator_state_storage_key(identity)?;
+			let storage_in_hex = client.query_delegator_state(&storage_key)?;
+
+			let delegator = DelegatorState::decode_delegator(&storage_in_hex)?;
+			let total = delegator.total;
+
+			total_staking_amount += total;
+		}
+
+		Ok(total_staking_amount / LIT_TOKEN_DECIMALS)
+	}
+
+	fn delegator_state_storage_key(identity: &Identity) -> Result<String> {
+		// encoded partial key: ParachainStaking DelegatorState
+		// 0xa686a3043d0adcf2fa655e57bc595a78131da8bc800de21b19b3ba9ed33cfacc
+		let params = "0xa686a3043d0adcf2fa655e57bc595a78131da8bc800de21b19b3ba9ed33cfacc";
+		let acc = identity
+			.to_account_id()
+			.ok_or(Error::RequestVCFailed(Assertion::LITStaking, ErrorDetail::ParseError))?;
+		let cocat = Twox64Concat::hash(acc.as_ref());
+
+		Ok(params.to_string() + &hex::encode(&cocat))
+	}
+
+	fn decode_delegator(storage_in_hex: &str) -> Result<Delegator<AccountId, Balance>> {
```

Comment:
We'll need to double check if `Balance` is parentchain balance, even if they eventually use the same type, it's always better to make it explicit.

Reply:
using ParentchainBalance instead.